### PR TITLE
DBT-1097 Feat/manual verification

### PR DESCRIPTION
**Pull request by kayiganwa04**
## Context

We are adding this form for making sure that we validate the document of the user of the user 

- [ ] we have to make sure we change the link in the make.com for production

## Test

- Go to this link http://localhost:3000/verification/45d4e2bdf03247bc8a08b9c9b4edb393

## Proof of Concept
image

### CODE REVIEW

File: src/app/verification/[id]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+          <br />
+          <p className="mb-6">
+            {t(
+              "sidebar.we_are_excited_to_help_you_plan_your_next_water_adventure"
+            )}
+          </p>
+          <p className="mb-6">{t("sidebar.sidebar_p1")}</p>
+          <p className="mb-6">{t("sidebar.sidebar_p2")}</p>
+        </div>{" "}
+      </Sidebar>
```
Comment (abelbordonado) [inline src/app/verification/[id]/page.tsx:73]: This is now in 2 places, we can do a component from it <BookingSideBar />

File: src/app/verify-document/[id]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+export default function VerifyDocument({ params }: { params: { id: string } }) {
+  const { t } = useTranslation();
+  const [error, setError] = useState<string>("");
+  const router = useRouter();
+  const id = useRef(params.id);
+  useEffect(() => {
+    const updateDocumentVerified = async () => {
+      const bookingInfo = new Booking({
+        DocumentsApproved: true,
```
Comment (abelbordonado) [inline src/app/verify-document/[id]/page.tsx:24]: Is this always true? is possible they say is not verified

File: src/components/booking/index.tsx [hunk 3]
```diff
@@ -101,6 +101,7 @@ export default function BookingComponent({
         booking={formData as unknown as Booking}
         onSuccess={() => router.replace("/success")}
         bookingId={id}
+        steps={["fuel", "sign", "validateFront", "validateBack", "uploadFrontIdImage", "uploadBackIdImage", "pay", "saveData", "notifyCustomer"]}
```
Comment (abelbordonado) [inline src/components/booking/index.tsx:104]: You need to merge with the parent PR, I added a step after `validateBack` that I was missing this morning (sorry!)

File: src/components/verification/index.tsx [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+    "Last Name": "",
+    "ID Number": "",
+    documentType: "National ID",
+    ID_Front_Picture: {} as File,
+    ID_Back_Picture: {} as File,
+  });
+
+  const submitVerificationForm = async () => {
+    saveModalRef.current?.start();
+    console.log(">>>>>>>>Info", data)
```
Comment (abelbordonado) [inline src/components/verification/index.tsx:45]: 🐌

File: src/components/verification/index.tsx [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+    <>
+      <SaveBooking
+        ref={saveModalRef}
+        formData={data as BookingFormData}
+        setFormData={setData}
+        boat={boatInfo}
+        booking={data as unknown as Booking}
+        onSuccess={() => router.replace("/success")}
+        bookingId={bookingId}
+        steps={["validateFront", "validateBack", "uploadFrontIdImage", "uploadBackIdImage", "pay", "saveData", "notifyCustomer"]}
```
Comment (abelbordonado) [inline src/components/verification/index.tsx:69]: We dont need to pay here, can remove pay step

File: src/components/verification/index.tsx [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+                    onChange={() =>
+                      setData({ ...data, documentType: "National ID" })
+                    }
+                    required
+                    className="w-4 h-4 text-blue-600 bg-gray-100 border-gray-300 dark:ring-offset-gray-800 dark:bg-gray-700 dark:border-gray-600"
+                  />
+                  <label className="ms-2 md:text-base text-sm  text-black">
+                    {t("input.national_id")}
+                  </label>
+                </div>
```
Comment (abelbordonado) [inline src/components/verification/index.tsx:129]: This input shall be common component, there is at least 6 on the code I think so far

File: src/app/verify-document/[id]/page.tsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        setError(t("error.error_booking_details"));
+        return;
+      }
+
+      const boatDetails = await getBoatInfo(data.Boat[0])
+
+      if (!boatDetails) {
+        setError(t("error.error_boat_details"));
+        return;
+      }
```
Comment (abelbordonado) [inline src/app/verify-document/[id]/page.tsx:49]: We dont need boat info here I think

File: src/components/verification/index.tsx (hunk 5 above)
Comment (abelbordonado) [inline src/components/verification/index.tsx:69]: dont use saveData (you are having issues right? 
Save data create a slot an other stuff

Lets create a different step for it with a different name

File: src/app/verify-document/[id]/page.tsx [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+  const { t } = useTranslation();
+  const [error, setError] = useState<string>("");
+  const router = useRouter();
+  const id = useRef(params.id);
+
+  const approved = useSearchParams().get("approved") || ""
+
+  useEffect(() => {
+    if (!["true", "false"].includes(approved)) {
+      return
```
Comment (abelbordonado) [inline src/app/verify-document/[id]/page.tsx:26]: Maybe redirect here to the home or set an error (

File: src/components/booking/partial/steps-actions.tsx [hunk 9]
```diff
@@ -31,6 +30,7 @@ export const steps: any = [
   "uploadBackIdImage",
   "pay",
   "saveData",
+  "saveDataOnValidation",
```
Comment (abelbordonado) [inline src/components/booking/partial/steps-actions.tsx:33]: Im not sure about here.
We are suppose to have 2 set of steps list
`stepsbooking`
`stepsReuploadIdentity` 
Here it seems you will break the booking page with an extra not required step

File: src/components/booking/partial/steps-actions.tsx [hunk 10]
```diff
@@ -356,6 +355,58 @@ export const stepsActions = ({
     },
   };
 
+  const saveDataOnValidation = {
+    execute: async (formData: BookingFormData, boat: Boat) => {
```
Comment (abelbordonado) [inline src/components/booking/partial/steps-actions.tsx:359]: `BookingFormData` 
This is a different form, maybe we can use different type  (Idea, but you may get confuse with typing

File: src/components/booking/partial/steps-actions.tsx [hunk 11]
```diff
@@ -356,6 +355,58 @@ export const stepsActions = ({
...
+
+      const {
+        ID_Back_Picture,
+        ID_Front_Picture,
+        ...bookingData
+      } = formData;
+
+
+      const bookingInfo = new Booking({
+        ...bookingData,
```
Comment (abelbordonado) [inline src/components/booking/partial/steps-actions.tsx:374]: Im not sure why we need this
I prefer just to see what we are going to update
 - name
  - last name
  - front
   - back
    - approved or not
    
    Only this 5-6 params

**Comment by abelbordonado** [comment]
image

Second document is missing - I added.

[ ] Test that uploading 1 document doesn't fail now that we try to send 2 documents

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
Test again the scenario when I submit the passport for second time the email of them also receive a email for checking

File: src/components/booking/partial/steps-actions.tsx (hunk 10 above)
Comment (kayiganwa04) [reply src/components/booking/partial/steps-actions.tsx:342]: I just created another type similar to the booking and removed unused attributes

**Pull request merged**